src/Microsoft.ML.Transforms/CategoryImputer.cs

```diff
@@ -0,0 +1,1824 @@
+                ctx.Writer.Write(data.Length);
+                ctx.Writer.Write(data);
+            }
+
+        }
+
+        public void Dispose()
+        {
+            //cppTransformer.Dispose();
+        }
+
+        #region C++ Safe handle classes
+
+        internal class TransformerEstimatorSafeHandler : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            private DestroyCppTransformerEstimator _destroyCppTransformerEstimator;
+            public TransformerEstimatorSafeHandler(IntPtr handle, DestroyCppTransformerEstimator destroyCppTransformerEstimator) : base(true)
+            {
+                SetHandle(handle);
+                _destroyCppTransformerEstimator = destroyCppTransformerEstimator;
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here. There shoudln't ever be one though.
+                return _destroyCppTransformerEstimator(handle, out IntPtr errorHandle);
+            }
+        }
+
+        // Destroying saved data is always the same.
+        internal class SaveDataSafeHandler : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            private IntPtr _dataSize;
+
+            [DllImport("Featurizers", EntryPoint = "CatImputerFeaturizer_string_DestroyTransformerSaveData", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+            private static extern bool DestroyTransformerSaveDataCpp(IntPtr buffer, IntPtr bufferSize, out IntPtr error);
+
+            public SaveDataSafeHandler(IntPtr handle, IntPtr dataSize) : base(true)
+            {
+                SetHandle(handle);
+                _dataSize = dataSize;
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here.  There shoudln't ever be one though.
+                return DestroyTransformerSaveDataCpp(handle, _dataSize, out IntPtr errorHandle);
+            }
+        }
+
+        internal class TransformedDataSafeHandler : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            private DestroyTransformedDataCpp _destroySaveDataHandler;
+            private IntPtr _dataSize;
+
+            public TransformedDataSafeHandler(IntPtr handle, IntPtr dataSize, DestroyTransformedDataCpp destroyCppTransformerEstimator) : base(true)
+            {
+                SetHandle(handle);
+                _dataSize = dataSize;
+                _destroySaveDataHandler = destroyCppTransformerEstimator;
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here.  There shoudln't ever be one though.
+                return _destroySaveDataHandler(handle, _dataSize, out IntPtr errorHandle);
+            }
+        }
+
+        [DllImport("Featurizers", EntryPoint = "GetErrorInfoString", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
```

Comment:
These new transforms should be split out of the main `Microsoft.ML` nuget package. Since you are adding them to the main nuget package, EVERY application that uses ML.NET will have a dependency on these native assemblies, even if they don't use these new transforms.

See what we did with the transforms that depend on MKL - we split them into their own nuget package, so only people who needed those components would be forced to bring MKL along with their application.

https://www.nuget.org/packages/Microsoft.ML.Mkl.Components/